Sources/SwiftSyntax/SyntaxNodes.swift.gyb.template

```diff
@@ -122,14 +122,33 @@ public struct ${node.name}: ${base_type}Protocol, SyntaxHashable {
 %       end
       return ${child.type_name}(childData!)
     }
+%       if not node.must_uphold_invariant:
     set(value) {
-%       if node.must_uphold_invariant:
-      self = with${child.name}(value)!
-%       else:
       self = with${child.name}(value)
+    }
 %       end
+  }
+%       if node.must_uphold_invariant:
+
+  public enum ${child.name}Error: Error, CustomStringConvertible {
+    case invalid(${child.swift_name}: ${ret_type})
+
+    public var description: String {
+      switch self {
+      case .invalid(let ${child.swift_name}):
+        return "attempted to use setter with invalid ${child.name} \"\(${child.swift_name})\""
+      }
+    }
+  }
+
+  mutating public func set${child.name}(_ ${child.swift_name}: ${ret_type}) throws {
+    if let tokenSyntax = with${child.name}(${child.swift_name}) {
```

Comment:
I think technically this variable shouldn’t be called `tokenSyntax` because the child doesn’t need to be a token (although it is in both of our cases). I would prefer something like `childSyntax` or something along those lines.

Reply:
Done.